src/main/java/org/jenkinsci/plugins/registry/notification/webhook/acr/ACRWebHookPayload.java

```diff
@@ -0,0 +1,53 @@
+package org.jenkinsci.plugins.registry.notification.webhook.acr;
+
+import net.sf.json.JSONObject;
+import org.jenkinsci.plugins.registry.notification.webhook.WebHookPayload;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import javax.annotation.Nonnull;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Partially implements https://docs.microsoft.com/en-us/azure/container-registry/container-registry-webhook-reference.
+ */
+public class ACRWebHookPayload extends WebHookPayload {
+    private static final Logger logger = Logger.getLogger(ACRWebHookPayload.class.getName());
+
+    public enum Action {
+        PUSH("push");
+        private String name;
+
+        private Action(String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return this.name;
+        }
+    }
+
+    public ACRWebHookPayload(@Nonnull final JSONObject data) {
+        setData(data);
+        setJson(data.toString());
+
+        if (Action.PUSH.getName().equals(data.optString("action"))) {
+            final JSONObject event = data;
+            final String host = event.getJSONObject("request").getString("host");
+            final String url = String.format("%s/%s",
```

Comment:
Seems like overkill for just concatenating three strings together. You have for example `hudson.Functions.joinPath` or `StringBuilder` or simple `+` that would be more efficient.

Reply:
Ah, stupid old habit :) I will change that to using `StringBuilder` ASAP. Thanks :)